.github/workflows/main.yml

```diff
@@ -42,3 +42,29 @@ jobs:
           file: ./Dockerfile
           push: true
           tags: ${{ secrets.DOCKER_HUB_USERNAME }}/prjctr02-infrastructure:latest
+
+  # For deployment on cluster we'll use self-made recipes :)
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install aws cli
+        run: |
+          which aws
+          aws --version
+          mkdir -p ~/.aws
+          touch ~/.aws/credentials
+          echo "[default]" >> ~/.aws/credentials
+          echo "aws_access_key_id = ${{ secrets.AWS_ACCESS_KEY_ID }}" >> ~/.aws/credentials
+          echo "aws_secret_access_key = ${{ secrets.AWS_SECRET_ACCESS_KEY }}" >> ~/.aws/credentials
+          cat ~/.aws/credentials
+
+      - name: Run Kubectl
+        run: |
+          mkdir -p ~/.kube
+          echo "${{ secrets.KUBE_CONFIG_DATA }}" | base64 -d >> ~/.kube/config
+          curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+          chmod +x ./kubectl
+          ./kubectl version --client
+          ./kubectl apply -f .cluster/service-conf.yaml
```

Comment:
could we also create job, pod, deployment with this cluster?

Reply:
`.cluster/service-conf.yaml` file already contains config for service and deployment. Execution of deployment section automatically creates underlying pod for the deployment. So, if I understand the task properly there is no need create separate pods and deployments.

But it is possible to have separate pods and deployments created within `.github/workflows/main.yml` file.
It may look like this:

```suggestion
          mkdir -p ~/.kube
          echo "${{ secrets.KUBE_CONFIG_DATA }}" | base64 -d >> ~/.kube/config
          curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
          chmod +x ./kubectl
          ./kubectl version --client
          ./kubectl apply -f .cluster/pod-conf.yaml
          ./kubectl apply -f .cluster/deployment-conf.yaml
          ./kubectl apply -f .cluster/job-conf.yaml
          ./kubectl apply -f .cluster/service-conf.yaml
```
Is it what you meant in your question?
Before applying those configs I should check container names conflicts, right?

```yaml
    spec:
      containers:
        - name : web-app-container    # <-- is this name must be unique across all applied configs?
          image: yevhenk10s/prjctr02-infrastructure:latest
          ports:
            - containerPort: 5000
```

Reply:
<-- is this name must be unique across all applied configs?

nope. this should be unique inside of one spec, so all good now